cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -42,61 +50,107 @@ import (
 // /etc/kubernetes/{admin,kubelet}.conf exist but the CA cert doesn't match what's in the pki dir => error
 // /etc/kubernetes/{admin,kubelet}.conf exist but not certs => certs will be generated and conflict with the kubeconfig files => error
 
-// CreateAdminAndKubeletKubeConfig is called from the main init and does the work for the default phase behaviour
-func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) error {
-
-	// Try to load ca.crt and ca.key from the PKI directory
-	caCert, caKey, err := pkiutil.TryLoadCertAndKeyFromDisk(pkiDir, kubeadmconstants.CACertAndKeyBaseName)
-	if err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig; the CA files couldn't be loaded: %v", err)
+// CreateInitKubeConfigFiles is called from the main init and does the work for the default phase behaviour
+func CreateInitKubeConfigFiles(masterEndpoint, pkiDir, outDir string) error {
+
+	// Create a lightweight specification for what the files should look like
+	filesToCreateFromSpec := map[string]KubeConfigProperties{
+		kubeadmconstants.AdminKubeConfigFileName: {
+			ClientName:      "kubernetes-admin",
+			APIServer:       masterEndpoint,
+			CertDir:         pkiDir,
+			Organization:    []string{kubeadmconstants.MastersGroup},
+			MakeClientCerts: true,
+		},
+		kubeadmconstants.KubeletKubeConfigFileName: {
+			ClientName:      "system:node:<hostname>",
```

Comment:
This is just for the master node's client cert, the hostname here plays no big role.

However, this should obviously be `fmt.Sprintf("system:node:%s", hostname)`, seems like that part didn't get pushed earlier. Now it is